cosigner/cosigner.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 OpenPubkey
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package cosigner
+
+import (
+	"crypto"
+	"encoding/json"
+
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/openpubkey/openpubkey/pktoken"
+)
+
+type Cosigner struct {
+	Alg    jwa.KeyAlgorithm
+	Signer crypto.Signer
+}
```

Comment:
Wait, we have a `Cosigner` and a `CosignerProvider`? Can you explain why we need both?

Reply:
Cosigner and AuthCosigner which does the Cosigning, CosignerProvider is the client which talks to the Cosigner. I debated CosignerProvider vs CosignerClient, but given that we call the OpenIDClient the OpenIDProvider it seemed like a more natural name.